packages/server/src/fhir/repo.ts

```diff
@@ -218,21 +218,27 @@ export interface ResendSubscriptionsOptions extends InteractionOptions {
 export class Repository extends FhirRepository<PoolClient> implements Disposable {
   private readonly context: RepositoryContext;
   private conn?: PoolClient;
+  private readonly disposable: boolean = true;
   private transactionDepth = 0;
   private closed = false;
   mode: RepositoryMode;
 
   private preCommitCallbacks: (() => Promise<void>)[] = [];
   private postCommitCallbacks: (() => Promise<void>)[] = [];
 
-  constructor(context: RepositoryContext) {
+  constructor(context: RepositoryContext, conn?: PoolClient) {
```

Comment:
Does anything need to change with the `clone` method below to account for a passed in `conn` to future proof it? I think I'd expect `conn` to be reused as well in the clone, but that's just my first take on it. Perhaps `clone` should throw if `this.disposable` is false?